app/frontend/wallet/shelley/helpers/poolCertificateUtils.ts

```diff
@@ -10,68 +34,64 @@ const enum PoolParamsByteLengths {
   METADATA_HASH = 32,
 }
 
-export type PoolOwnerParams = {
-  stakingPath?: any //BIP32Path
+type TxStakepoolOwner = {
   stakingKeyHashHex?: string
 }
 
-export type SingleHostIPRelay = {
-  portNumber?: number
-  ipv4?: string
-  ipv6?: string
-}
+export type TxStakepoolRelay = SingleHostIPRelay | SingleHostNameRelay | MultiHostNameRelay
 
-export type SingleHostNameRelay = {
-  portNumber?: number
-  dnsName: string
+type SingleHostIPRelay = {
+  type: TxRelayType.SINGLE_HOST_IP
+  params: {
+    portNumber?: number
+    ipv4?: string
+    ipv6?: string
+  }
 }
 
-export type MultiHostNameRelay = {
-  dnsName: string
+type SingleHostNameRelay = {
+  type: TxRelayType.SINGLE_HOST_NAME
+  params: {
+    portNumber?: number
+    dnsName: string
+  }
 }
 
-export type RelayParams = {
-  type: number // single host ip = 0, single hostname = 1, multi host name = 2
-  params: SingleHostIPRelay | SingleHostNameRelay | MultiHostNameRelay
+type MultiHostNameRelay = {
+  type: TxRelayType.MULTI_HOST_NAME
+  params: {
+    dnsName: string
+  }
 }
 
-export type PoolMetadataParams = {
+type TxStakepoolMetadata = {
   metadataUrl: string
   metadataHashHex: string
 }
 
-export type Margin = {
+type TxStakepoolMargin = {
   numeratorStr: string
   denominatorStr: string
 }
 
-export type PoolParams = {
-  poolKeyHashHex: string
-  vrfKeyHashHex: string
-  pledgeStr: string
-  costStr: string
-  margin: Margin
-  rewardAccountHex: string
-  poolOwners: Array<PoolOwnerParams>
-  relays: Array<RelayParams>
-  metadata: PoolMetadataParams
+// TODO: fix when migrating to BigInt
+export const validateNumber = (number: number | BigInt, variableName: string): number => {
```

Comment:
sorry for not being clear in the previous comment - the validator is IMHO OK to expect `number: any` (or rather `value: any`) at the input (as the input is unvalidated), but I'd rather add runtime assertions on the input being an instance of number and checking its range.

Also, expecting an instance of BigInt only to reject it if it exceeds MAX_SAFE_INTEGER seems a bit strange and if really needed, I'd rather explicitly acknowledge with a comment that even though some values can be BigInt, the codebase isn't fully migrated to BigInt, that's why we restrict the range even for those